### ISSUE TITLE
lintDirtyModulesOnly:true does NOT skip lint on start (webpack --watch or webpack serve)

### ISSUE THREAD

**Issue by attekemppila** [webpack-contrib/eslint-webpack-plugin#60]
- Operating System: Linux
- Node Version: 10.23.0
- NPM Version: 6.14.8
- webpack Version: 4.44.2
- eslint-webpack-plugin Version: 2.4.1

### Expected Behavior

With `lintDirtyModulesOnly: true` and running `webpack --watch` or `webpack serve`, I expect lint to be skipped at the start. And only happen after I edit a file and webpack re-compiles.

### Actual Behavior

Lint warnings and errors are reported at start-up.

```
$ npm run webpack:serve

> eslint-webpack-plugin-test@1.0.0 webpack:serve /home/atte/eslint-webpack-plugin-test
> cross-env NODE_ENV=development webpack serve

ℹ ｢wds｣: Project is running at http://localhost:8080/
ℹ ｢wds｣: webpack output is served from /
ℹ ｢wds｣: Content not from webpack is served from /home/atte/eslint-webpack-plugin-test
⚠ ｢wdm｣: Hash: a74bdf32336d9eca0380
Version: webpack 4.44.2
Time: 835ms
Built at: 01/05/2021 5:11:53 PM
  Asset     Size  Chunks             Chunk Names
main.js  369 KiB    main  [emitted]  main
Entrypoint main = main.js
[0] multi (webpack)-dev-server/client?http://localhost:8080 ./src 40 bytes {main} [built]
[./node_modules/ansi-html/index.js] 4.16 KiB {main} [built]
[./node_modules/html-entities/lib/index.js] 449 bytes {main} [built]
[./node_modules/loglevel/lib/loglevel.js] 8.65 KiB {main} [built]
[./node_modules/url/url.js] 22.8 KiB {main} [built]
[./node_modules/webpack-dev-server/client/index.js?http://localhost:8080] (webpack)-dev-server/client?http://localhost:8080 4.29 KiB {main} [built]
[./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.51 KiB {main} [built]
[./node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 1.53 KiB {main} [built]
[./node_modules/webpack-dev-server/client/utils/createSocketUrl.js] (webpack)-dev-server/client/utils/createSocketUrl.js 2.91 KiB {main} [built]
[./node_modules/webpack-dev-server/client/utils/log.js] (webpack)-dev-server/client/utils/log.js 964 bytes {main} [built]
[./node_modules/webpack-dev-server/client/utils/reloadApp.js] (webpack)-dev-server/client/utils/reloadApp.js 1.59 KiB {main} [built]
[./node_modules/webpack-dev-server/client/utils/sendMessage.js] (webpack)-dev-server/client/utils/sendMessage.js 402 bytes {main} [built]
[./node_modules/webpack-dev-server/node_modules/strip-ansi/index.js] (webpack)-dev-server/node_modules/strip-ansi/index.js 161 bytes {main} [built]
[./node_modules/webpack/hot sync ^\.\/log$] (webpack)/hot sync nonrecursive ^\.\/log$ 170 bytes {main} [built]
[./src/index.js] 30 bytes {main} [built]
    + 19 hidden modules

WARNING in 
/home/atte/eslint-webpack-plugin-test/src/index.js
  1:1  warning  Expected indentation of 0 spaces but found 2  indent
  1:3  warning  Unexpected console statement                  no-console

✖ 2 problems (0 errors, 2 warnings)
  0 errors and 1 warning potentially fixable with the `--fix` option.

ℹ ｢wdm｣: Compiled with warnings.
```

### How Do We Reproduce?

Example repo: https://github.com/attekemppila/eslint-webpack-plugin-test

```sh
npm run lint
# eslint warnings in src/index.js

npm run webpack:watch
# eslint warnings in src/index.js - reported at start-up, but should not be

npm run webpack:serve
# eslint warnings in src/index.js - reported at start-up, but should not be
```

**Comment by alexander-akait** [comment]
It should not skip on any start, because you can have problems with code

**Comment by attekemppila** [comment]
> It should not skip on any start, ...

That's not what the documentation says:

https://github.com/webpack-contrib/eslint-webpack-plugin#lintdirtymodulesonly

> Lint only changed files, skip lint on start.

So, there's a bug in code or in documentation.

I would say that the bug is in the code. I mean, what is the purpose of `lintDirtyModulesOnly` otherwise? Or am I missing something here?

Also, that's how [lintDirtyModulesOnly](https://webpack.js.org/plugins/stylelint-webpack-plugin/#lintdirtymodulesonly) seems to work in [stylelint-webpack-plugin](https://github.com/webpack-contrib/stylelint-webpack-plugin/) for example. If `lintDirtyModulesOnly: true`, errors are not reported at start-up, but after re-compile (if a module is dirty, I assume).

**Comment by boroth** [comment]
I would also recommend either clearing up the documentation or changing the behavior, it's a bit confusing.

In my case, I'm also seeing that eslint is being run for my entire project when a single file changes, instead of running just for the updated file/module. @attekemppila are you seeing the same?

**Comment by attekemppila** [comment]
> In my case, I'm also seeing that eslint is being run for my entire project when a single file changes, instead of running just for the updated file/module. @attekemppila are you seeing the same?

Yes, same here.
